### PR TITLE
[onert-micro] Introduce eval driver

### PR DESCRIPTION
**Pull request by BalyshevArtem**
This pr adds eval driver for onert-micro.

ONE-DCO-1.0-Signed-off-by: Artem Balyshev <a.balyshev@samsung.com>

### CODE REVIEW

File: onert-micro/eval-driver/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+set(SRCS_EVAL_TESTER Driver.cpp)
+
+add_executable(onert_micro_eval_driver ${SRCS_EVAL_TESTER})
+
+set(READER_SUFFIX "_driver")
+
+add_subdirectory(${NNAS_PROJECT_SOURCE_DIR}/onert-micro/luci-interpreter ${CMAKE_CURRENT_BINARY_DIR}/luci-interpreter)
```
Comment (binarman) [inline onert-micro/eval-driver/CMakeLists.txt:7]: notpick:
```suggestion
add_subdirectory(${NNAS_PROJECT_SOURCE_DIR}/onert-micro/luci-interpreter)
add_subdirectory(${CMAKE_CURRENT_BINARY_DIR}/luci-interpreter)
```

Comment (binarman) [reply onert-micro/eval-driver/CMakeLists.txt:7]: sorry, I messed up :)

File: onert-micro/eval-driver/Driver.cpp [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    else
+    {
+      auto shape_str = std::to_string(output_tensor->shape().dim(0));
+      for (int j = 1; j < output_tensor->shape().num_dims(); j++)
+      {
+        shape_str += ",";
+        shape_str += std::to_string(output_tensor->shape().dim(j));
+      }
+      writeDataToFile(std::string(output_file) + std::to_string(i) + ".shape", shape_str.c_str(),
+                      shape_str.size());
```
Comment (binarman) [inline onert-micro/eval-driver/Driver.cpp:157]: maybe something like this?

```suggestion
      const auto tensor_shape_file = std::string(output_file) + std::to_string(i) + ".shape";
      writeDataToFile(tensor_shape_file, shape_str.c_str(), shape_str.size());
```

File: onert-micro/eval-driver/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,12 @@
+set(SRCS_EVAL_TESTER Driver.cpp)
+
+add_executable(onert_micro_eval_driver ${SRCS_EVAL_TESTER})
+
+set(READER_SUFFIX "_driver")
```
Comment (binarman) [inline onert-micro/eval-driver/CMakeLists.txt:5]: Could you add some comment here?
Like "This variable is needed to separate standalone interpreter libraries from the libraries used in driver"

**Review by binarman — APPROVED** [review]
LGTM,
Take a look at comments and let me know if you are going to change anything here.

File: onert-micro/eval-driver/Driver.cpp [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+    throw std::runtime_error("Cannot open file \"" + filename + "\".\n");
+  if (fs.write(data, data_size).fail())
+  {
+    throw std::runtime_error("Failed to write data to file \"" + filename + "\".\n");
+  }
+}
+
+size_t getTensorSize(const luci_interpreter::Tensor *tensor)
+{
+  uint32_t tensor_size = luci_interpreter::size(tensor->element_type());
```
[the remark this answers is not in the document]
Comment (BalyshevArtem) [reply onert-micro/eval-driver/Driver.cpp]: Done

File: onert-micro/eval-driver/CMakeLists.txt (hunk 3 above)
Comment (BalyshevArtem) [reply onert-micro/eval-driver/CMakeLists.txt:6]: Done

**Review by binarman — APPROVED** [review]

**Pull request merged**